kong/db/migrations/core/016_280_to_300.lua

```diff
@@ -388,40 +358,31 @@ local function p_update_cache_key(connector)
   return true
 end
 
+local function ensure_empty_vaults_tables(connector)
+  local ensure_table_is_empty = function (table)
+    local res, err = connector:query("SELECT * FROM " .. table)
+    if err then
+      -- Assume that the error is about the missing table, which is OK
+      return true
+    end
+    if #res > 0 then
+      return nil, "Cannot perform database upgrade with data in " .. table .. " table.  Please delete all rows from it and retry"
+    end
+  end
```

Comment:
When we want to communicate that a function  like `ensure_empty_vaults_table` has some "private aux functions or variables" we usually use a `do ... end` block (slightly abusing the fact that Lua doesn't care about newlines):

```
local function ensure_empty_vaults_tables do

  local ensure_table_is_empty = function (table, connector)
    local res, err = connector:query("SELECT * FROM " .. table)
    if err then
      -- Assume that the error is about the missing table, which is OK
      return true
    end
    if #res > 0 then
      return nil, "Cannot perform database upgrade with data in " .. table .. " table.  Please delete all rows from it and retry"
    end
  end
  
  ensure_empty_vaults_tables = function(connector)
    ...
  end
end
```

Reply:
I don't quite understand this comment - I put the `ensure_table_is_empty` function into the lexical scope of `ensure_empty_vaults_tables`, which makes it only accessible in that function, isn't that right?  What you've shown in your example puts the `ensure_table_is_empty` into the lexical scope of the `do` block.  How is that better?  I notice that I have to spell out `ensure_empty_vaults_tables` twice in the definition, which I find to be less clear, but maybe there is something else in there that I don't see?